### PR TITLE
Feature/vhris 3

### PR DESCRIPTION
**Pull request by mikelcop-condebender**

### CODE REVIEW

File: src/employees/entities/employee.entity.ts [hunk 1]
```diff
@@ -7,53 +7,53 @@ export type EmployeeDocument = Employee & Document;
...
+  @Prop({ required: true })
   lastName: string;
   @Prop()
   middleName: string;
-  @Prop()
-  fsCode: string;
-  @Prop()
-  bioCode: string;
-  @Prop()
+  @Prop({ required: true })
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts:26]: New columns: 

1. suffix, not required
2. citizenship, required

File: src/employees/dto/create-employee.dto.ts [hunk 2]
```diff
@@ -1,161 +1,180 @@
...
-  civilStatus?: string;
+  @ApiProperty({ required: true })
+  @IsEnum(CivilStatusEnum)
+  @Transform((param) => param.value.toUpperCase())
+  civilStatus: CivilStatusEnum;
+
+  @ApiProperty()
+  @IsEnum(ReligionEnum)
+  @Transform((param) => param.value.toUpperCase())
+  religion: ReligionEnum;
```
Comment (lianiemiagdan) [inline src/employees/dto/create-employee.dto.ts:109]: New column: **citizenship**

File: src/employees/entities/employee.entity.ts [hunk 3]
```diff
@@ -7,53 +7,53 @@ export type EmployeeDocument = Employee & Document;
...
   gender: string;
-  @Prop()
+  @Prop({ required: true })
   birthDate: Date;
-  @Prop()
-  age: number;
-  @Prop()
+  @Prop({ required: true })
   contactNumber: string;
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts]: Change to personalContactNumber - required
New column: **companyContactNumber**

File: src/employees/entities/employee.entity.ts [hunk 4]
```diff
@@ -82,27 +82,21 @@ export class Employee {
...
   @Prop()
   cocNo: string;
   @Prop()
-  vaccineStatus: string;
-  @Prop()
-  digitalBulletin: string;
-  @Prop()
-  viberNumber: string;
-  @Prop()
   vpdcEmail: string;
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts]: Remove column **vpdcEmail**

File: src/employees/interface/employee.interface.ts [hunk 5]
```diff
@@ -1,49 +1,45 @@
...
+  NumberOfDependents: number;
   sss: string;
   philHealth: string;
-  pagIbig: string;
+  pagIbig?: string;
   tin: string;
   city: string;
   zipCode: string;
   region: string;
   address: string;
```
Comment (lianiemiagdan) [inline src/employees/interface/employee.interface.ts]: 1. Change to **permanentResidenceAddress**.
2. New column **presentResidenceAddress**

File: src/employees/interface/employee.interface.ts [hunk 6]
```diff
@@ -1,49 +1,45 @@
...
   civilStatus: string;
-  NumberOfDependents: string;
+  religion: string;
+  NumberOfDependents: number;
   sss: string;
   philHealth: string;
-  pagIbig: string;
+  pagIbig?: string;
   tin: string;
   city: string;
```
Comment (lianiemiagdan) [inline src/employees/interface/employee.interface.ts]: 1. Change to **presentCity**.
2. New column **permanentCity**

File: src/employees/interface/employee.interface.ts [hunk 7]
```diff
@@ -1,49 +1,45 @@
...
   zipCode: string;
```
Comment (lianiemiagdan) [inline src/employees/interface/employee.interface.ts]: 1. Change to **presentZipCode**
2. New column **permanentZipCode**

File: src/employees/interface/employee.interface.ts [hunk 8]
```diff
@@ -1,49 +1,45 @@
...
   region: string;
```
Comment (lianiemiagdan) [inline src/employees/interface/employee.interface.ts]: 1. Change to **presentRegion**
2. New column **permanentRegion**

File: src/employees/interface/employee.interface.ts [hunk 9]
```diff
@@ -1,49 +1,45 @@
...
   address: string;
-  course: string;
-  educationalAttainment: string;
-  schoolAttended: string;
+  course?: string;
```
Comment (lianiemiagdan) [inline src/employees/interface/employee.interface.ts]: Remove column **course**

File: src/employees/interface/employee.interface.ts [hunk 10]
```diff
@@ -1,49 +1,45 @@
...
+  educationalAttainment?: string;
+  schoolAttended?: string;
```
Comment (lianiemiagdan) [inline src/employees/interface/employee.interface.ts]: Remove column **schoolAttended**

File: src/employees/interface/employee.interface.ts [hunk 11]
```diff
@@ -1,49 +1,45 @@
...
   tin: string;
   city: string;
   zipCode: string;
   region: string;
   address: string;
-  course: string;
-  educationalAttainment: string;
-  schoolAttended: string;
+  course?: string;
+  educationalAttainment?: string;
```
Comment (lianiemiagdan) [inline src/employees/interface/employee.interface.ts]: 1. Remove column **educationalAttainment**.
2. New columns:
- highestEducationalAttainment - with enums - check HRIS Database Structure List of Values sheet
- elementaryYrFrom
- elementaryYrTo
- elementarySchoolAndAddress
- elementaryHonors
- secondaryYrFrom
- secondaryYrTo
- secondarySchoolAndAddress
- secondaryHonors
- tertiaryYrFrom
- tertiaryYrTo
- tertiarySchoolAndAddress
- tertiaryDegree
- tertiaryHonors
- postGradYrFrom
- postGradYrTo
- postGradSchoolAndAddress
- postGradDegree
- postGradHonors
- othersYrFrom
- othersYrTo
- othersSchoolAndAddress
- othersDegree
- othersHonors

File: src/employees/entities/employee.entity.ts [hunk 12]
```diff
@@ -7,53 +7,53 @@ export type EmployeeDocument = Employee & Document;
...
+  @Prop({ required: true })
   taxExemption: string;
   @Prop({ required: true, unique: true })
-  email: string;
-  @Prop()
+  companyEmail: string;
+  @Prop({ required: true, unique: true })
+  personalEmail: string;
+  @Prop({ required: true })
   backAccountNo: string;
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts]: Change to payrollBankAccount, type is object
Sample: {
"accountName": "",
"accountNumber": "",
"bankName": "",
"bankBranch": "",
}

File: src/employees/entities/employee.entity.ts [hunk 13]
```diff
@@ -82,27 +82,21 @@ export class Employee {
...
+  @Prop({ required: true })
   emergencyContactPerson: string;
-  @Prop()
+  @Prop({ required: true })
   emergencyAddress: string;
-  @Prop()
+  @Prop({ required: true })
   emergencyContactNo: string;
+  @Prop()
+  password?: string;
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts]: 1. Remove column **password**.
2. New column **employmentRecords** - array of objects [{
"yrFrom": 2018,
"yrTo": 2022,
"companyName": "Company ABC",
"positionHeld": "Web Developer"
}]
3. New column **govtProfExamsPassed** - array of objects [{
"examTitle": "Board Exam",
"dateTaken": "02/20/2018",
"Rating": "81%"
}]
4. New column **licensesCertifications** - array of objects [{
"name": "license/certification",
"authorizingEntity": "PRC",
"validUntil": "02/20/2025",
"licenseCertNo": ""
}]
5. New column **familyBackground** - array of objects [{
"name": "",
"relation": "Parents" | "Siblings" | "Spouse" | "Children",
"occupation": "",
"company": "",
"residence": ""
}]

File: src/employees/entities/employee.entity.ts [hunk 14]
```diff
@@ -82,27 +82,21 @@ export class Employee {
...
-  vaccineStatus: string;
-  @Prop()
-  digitalBulletin: string;
-  @Prop()
-  viberNumber: string;
-  @Prop()
   vpdcEmail: string;
-  @Prop()
+  @Prop({ required: true })
   emergencyContactPerson: string;
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts]: Remove column **emergencyContactPerson**

File: src/employees/entities/employee.entity.ts [hunk 15]
```diff
@@ -82,27 +82,21 @@ export class Employee {
...
-  @Prop()
+  @Prop({ required: true })
   emergencyAddress: string;
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts]: Remove column **emergencyAddress**

File: src/employees/entities/employee.entity.ts [hunk 16]
```diff
@@ -82,27 +82,21 @@ export class Employee {
...
-  @Prop()
+  @Prop({ required: true })
   emergencyContactNo: string;
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts]: 1. Remove column **emergencyContactNo**
2. Add new column: **emergencyContact** - array of objects - [{
"name": "", "address": "", "phoneNumber": ""
 }]

File: src/employees/entities/employee.entity.ts [hunk 17]
```diff
@@ -82,27 +82,21 @@ export class Employee {
...
   prcIdNo: string;
   @Prop()
-  noticeOffense: string;
-  @Prop()
   audit201: string;
   @Prop()
-  remarks: string;
+  notes: string;
   @Prop()
   cocNo: string;
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts]: Remove column **cocNo**

File: src/employees/entities/employee.entity.ts [hunk 18]
```diff
@@ -82,27 +82,21 @@ export class Employee {
   @Prop()
   prcIdNo: string;
   @Prop()
-  noticeOffense: string;
-  @Prop()
   audit201: string;
   @Prop()
-  remarks: string;
+  notes: string;
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts]: Remove column **notes**

File: src/employees/entities/employee.entity.ts [hunk 19]
```diff
@@ -82,27 +82,21 @@ export class Employee {
   @Prop()
   prcIdNo: string;
   @Prop()
-  noticeOffense: string;
-  @Prop()
   audit201: string;
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts]: Remove column **audit201**

File: src/employees/entities/employee.entity.ts [hunk 20]
```diff
@@ -82,27 +82,21 @@ export class Employee {
   @Prop()
   prcIdNo: string;
```
Comment (lianiemiagdan) [inline src/employees/entities/employee.entity.ts]: Remove column **prcIdNo**

File: src/employees/interface/employee.interface.ts [hunk 21]
```diff
@@ -1,49 +1,45 @@
 interface EmployeeI {
```
Comment (lianiemiagdan) [inline src/employees/interface/employee.interface.ts:5]: apply changes based on the comments on employe.entity.ts file

File: src/app.module.ts [hunk 22]
```diff
@@ -11,13 +11,23 @@ import { LeaveTypesModule } from './leave_types/leave_types.module';
...
 import { UserCredentialsModule } from './user_credentials/user_credentials.module';
 import { AuthModule } from './auth/auth.module';
 
 @Module({
   imports: [
+    MailerModule.forRoot({
+      transport: {
+        host: 'smtp.sendgrid.net',
+        auth: {
+          user: 'apikey',
```
Comment (lianiemiagdan) [inline src/app.module.ts]: hindi sya naka-variable kuya mike?

File: src/employees/entities/employee.entity.ts (hunk 1 above)
Comment (mikelcop-condebender) [reply src/employees/entities/employee.entity.ts:26]: This is done.

**Pull request merged**